### ISSUE TITLE
Cpu utilization jump to 100% spawning a web browser in a pygtk twisted application

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4646]
|<img alt="pimander's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| pimander reported|
|-|-|
|Trac ID|trac#4646|
|Type|enhancement|
|Created|2010-09-06 14:48:44Z|

Hi,
I'm developing a IRC client in python based on pygtk and twisted. I need to spawn a web browser to open a url displayed in the text window. The browser open the url but when I close it the cpu utilization of the application jump to 100%. 
In attachment there is a simple adaptation of your example IrcLogBot to run in a pygtk window, to use for test you must edit the executable path for a web browser (konqueror, firefox etc) and pass a irc server url as parameter. Clicking the button fire the browser. Closing the browser you can verify the cpu utilization.
This does't happen in a purely pygtk application. I know a web browser is a big beast, It spawns various threads and frequently the first only runs some thread and terminate soon.
The versions used are:
Python 2.6.5 - pygtk 2.16.0 - twisted 10.1.0

marco

Attachments:

* [GtkIrcLogBot.py](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/533/53389e591d829ff241412dd612bf77735e5acdfc/0ee9bd1e2870e193a4565af850a420e29f51162a.py) (6067 bytes) - added by pimander on 2010-09-06 14:49:27Z - running example of the issue

<details><summary>Searchable metadata</summary>

```
trac-id__4646 4646
type__enhancement enhancement
reporter__pimander pimander
priority__normal normal
milestone__ 
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1283784524000000 1283784524000000
changetime__1283790639000000 1283790639000000
version__None None
owner__pimander pimander

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun set owner to pimander|
|-|-|

This sounds a lot like [#3096](https://github.com/twisted/twisted/issues/3096).  Can you try the shorter example attached to that ticket and see if it produces the same CPU utilization misbehavior?  If it does, then I think you've run into the pygtk/pygobject bug described there and the proper fix is likely to upgrade your installs of those libraries.  I don't think anyone ever figured out a way to correct the behavior from outside (ie, there seems to be nothing Twisted can do about the problem).

**Comment by twisted-trac** [comment]
|<img alt="pimander's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_2"></a>|pimander commented|
|-|-|

Hi exarkun,
you are right and the developers of both pygobject and pygtk already fixed the issue in pygobject-2.21.0 and pygtk-2.17.0.
I will upgrade asap, sometimes a happy end happens :)

Thanks for your attention and regards.
marco

**Comment by twisted-trac** [comment]
|<img alt="pimander's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|pimander set status to `closed`|
|-|-|
